docs/source/typing_anti_pitch.rst

```diff
@@ -0,0 +1,80 @@
+.. _typing-anti-pitch:
+
+Reasons to avoid static type checking
+=====================================
+
+In the words of :pep:`484`::
+
+    It should also be emphasized that Python will remain a dynamically typed language, and the
+    authors have no desire to ever make type hints mandatory, even by convention.
+
+The idea that dynamism in Python is a strength of the language is reflected in the fact that
+Python's type system is gradual. See :pep:`483` for details, but the long and short of this is
+that you can add static types to your codebase only to the extent that you want to, and static
+type checkers are designed to be able to put up with this.
+
+It's also worth noting that "static type checking" encompasses a spectrum of possible degrees of
+strictness. On the one hand, you can set yourself up so that your type checker does almost nothing.
+On the other -- well, I love type checking, but I would quit Python if I had to enable all
+possible strictness checks that type checkers offer.
+
+Anway, with all that said, here's a list of possible reasons to not use static type checking
+in Python::
+
+* You simply don't want to. Python is a tool that is meant to serve you. Python is a big tent,
+  multi-paradigm language that generally allows you to do things in the way that best suits your
+  needs, as best determined by you.
+
+* The cost-benefit ratio isn't good enough. Pleasing static type checkers requires a non-zero amount
```

Comment:
One interesting aspect of this is if your project is based on an ecosystem that simply doesn't work well with typing. For example, someone using FastAPI and SQLAlchemy will have a different (better) cost-benefit calculus here than someone using Django and the Django ORM. I personally view this as a reason to switch to better typed libraries but I bet the majority of the community would hold a different view ;)

Reply:
> I personally view this as a reason to switch to better typed libraries

Unfortunately, and this is the root issue here, many people see it as a reason to ask[^1] library authors to add type annotations. Particularly in the case of specialised libraries where there may not *be* alternatives.

[^1]: And as is often the case with feature requests in open source, "ask" often ends up being more like "demand" or "pressure". This is especially true when dealing with people working on large commercial projects, which are often the projects that benefit most from type annotations 🙁

Reply:
Thanks, I added https://github.com/python/typing/pull/1477/commits/4a3a6ebd7044533d675173fde3562285331ab958 to better account for this